### ISSUE TITLE
PGAdmin Testing Investigation

### ISSUE THREAD

**Issue by BramGruneir** [cockroachdb/cockroach#26603]
While testing out PGAdmin's test suite, the following issues cause test failures.  Note that this is not a list of extensions to implement but a list of what fell out of my investigation into PGAdmin.  We will not support a good number of these.

The most important to unlock testing correctly is the correct full handling of schemas.

- CREATE SCHEMA
- CREATE CAST
- CREATE EVENT TRIGGER
- CREATE EXTENSION
- CREATE FOREIGN DATA WRAPPER
- CREATE SERVER
- CREATE USER MAPPING FOR
- CREATE TRUSTED PROCEDURAL LANGUAGE
- CREATE COLLATION
- CREATE DOMAIN
- CREATE FOREIGN TABLE
- CREATE TEXT SEARCH CONFIGURATION
- CREATE TEXT SEARCH DICTIONARY
- CREATE TEXT SEARCH TEMPLATE
- CREATE DATABASE WITH OWNER
- CREATE USER WITH NOLOGIN
- CREATE ROLE username LOGIN
- COMMENT ON DATABASE
- DROP TEXT SEARCH PARSER
- DO $$
- pg_terminate_backend()

**Comment by BramGruneir** [comment]
@awoods187 Can you look through this list and augment it with what's on our roadmap and what's not?

**Comment by knz** [comment]
I have created a table above with the detail of the features. I think most of them we will not implement anytime soon. It would be better for now to simply skip these tests.

**Comment by vy-ton** [comment]
@rafiss I don't believe PGAdmin's test suite is integrated into our CI but it could be one to prioritized. Would be interesting to compare a GUI test suite against the driver and ORM ones

**Comment by ozturkberkay** [comment]
Hi. I noticed that `pg_cancel_backend` is also not supported. Are there any alternatives for `pg_terminate_backend` and/or `pg_cancel_backend` in CockroachDB?

**Comment by vy-ton** [comment]
https://github.com/cockroachdb/cockroach/issues/41335 is coming in 22.1 as well
